### DISCUSSION TITLE
Create json from dsl

### DISCUSSION THREAD

**Discussion by SaschaWeisenberger** [structurizr/onpremises#53] [Q&A] [answered]
### Question

Hi Simon,

Short question:
I followed the instructions here https://structurizr.com/help/web-api to upload a workspace definition. I plan to do it from a ci/cd chain, where I only have access to the workspace.dsl -> no json. After the update there are no diagrams in the ui, I need to go to the editor, click save and then it works.
Is there a way to generate the json after the upload?

Thanks!

EDIT: The workspace json I get looks like this
```json
{
    "id": 1,
    "name": "Name",
    "description": "Description",
    "revision": 10,
    "lastModifiedDate": "2023-07-14T08:49:07Z",
    "properties": {
        "structurizr.dsl": "IWNvbnN0YW..."
    },
    "model": {},
    "views": {
        "configuration": {
            "branding": {},
            "styles": {},
            "terminology": {}
        }
    }
}
```

**Comment by simonbrowndotje** [comment]
> Is there a way to generate the json after the upload?

I don't really understand what you're asking I'm afraid.

**Comment by simonbrowndotje** [comment]
> After the update there are no diagrams in the ui, I need to go to the editor, click save and then it works.

This shouldn't be the case. Which editor are you referring to?

**Comment by SaschaWeisenberger** [comment]
I upload a json object that looks like
```json
{
    "id": 1,
    "name": "Name",
    "description": "Description",
    "properties": {
        "structurizr.dsl": "IWNvbnN0YW..."
    },
}
```
Which means: I only PUT the workspace dsl (as base64).
When I read the same workspace, the properties "model" and "views" are empty. I guess thats why there are no diagrams in the web view.
I activated the dsl editor in the web ui, when I open it and click save the json for model and views are generated.

I hope this makes it clear. 
Thanks in advance!

**Comment by simonbrowndotje** [comment, marked as answer]
> I only PUT the workspace dsl (as base64).

That explains it ... you're pushing a JSON file without any views defined, and loading the workspace via the editor will be creating a default set for you. Why can't you push the DSL file?

Edit: If you're not doing so, I'd recommend using the Structurizr CLI to upload your workspace by pushing the DSL file -> https://github.com/structurizr/cli/blob/master/docs/push.md

**Reply by SaschaWeisenberger** [reply]
I tried exactly that, until I realized that the web api expects a json and not a dsl file :D
But thanks for that hint! I will use the cli instead (I guess the cli tool has the "engine" inside to create the json from the dsl file, right?)

**Reply by simonbrowndotje** [reply]
> But thanks for that hint!

You're welcome.

> I guess the cli tool has the "engine" inside to create the json from the dsl file, right?

It does, yes.
